### PR TITLE
feat!: add TF 0.13 constraint and module attribution

### PR DESCRIPTION
**Pull request by bharathkkb**
- add module attribution
- add TF 0.13 constraint

### CODE REVIEW

**Comment by morgante** [comment]
@bharathkkb Looks like tests are failing.

**Comment by bharathkkb** [comment]
@morgante It was failing due to the 3rd party provider. I have updated to use the new source syntax for both 3rd party and google providers.

File: modules/core_project_factory/versions.tf [hunk 1]
```diff
@@ -15,12 +15,30 @@
...
+      source  = "hashicorp/null"
+      version = "~> 2.1"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 2.2"
+    }
+  }
+  provider_meta "google" {
+    module_name = "blueprints/terraform/terraform-google-project-factory:core_project_factory/v10.0.0"
```
Comment (morgante) [inline modules/core_project_factory/versions.tf:39]: We shouldn't use add the `provider_meta` for internal modules so let's drop this section.

**Review by morgante — CHANGES_REQUESTED** [review]

File: modules/core_project_factory/versions.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/core_project_factory/versions.tf:39]: sg; in that case should `shared_vpc` be an internal module or should we add a readme for it?

Comment (bharathkkb) [reply modules/core_project_factory/versions.tf]: done

**Review by morgante — APPROVED** [review]

**Comment by hrvolapeter** [comment]
@bharathkkb was there a reason for requiring google providers less than 4.0.0 ? 4.0.0 is out and this is preventing from upgrading to the latest version.

**Pull request merged**